### PR TITLE
[BUG] - Focus management of Tabs Bar is broken when it's scrollable (Issue/3342)

### LINKED ISSUE #3342: [BUG] - Focus management of `Tabs Bar` is broken when it's scrollable
### Bug Description
Focus management is broken after tabbing through the scrollable container of `Tabs Bar` component. This might be due that the `p-scroller` component get's an `tabindex="-1"` if the inner "scroll-area" get's blurred.

**Example video:**
https://github.com/porsche-design-system/porsche-design-system/assets/40788023/7e413822-167a-4851-9ff3-8216bb1e93c0

### Contact Details

Slack: https://porschedev.slack.com/archives/CT7AVHTTQ/p1718177906400209

### JS Framework

all

### Porsche Design System version

3.16.0-rc.0

### On which device(s) you can see the bug?

all

### In which browsers you can see the bug?

all

### Browser version

all

### Reproduction Example

Storefront: https://designsystem.porsche.com/latest/components/tabs-bar/examples#gradient-color

### Steps to reproduce the behavior

1. Go to https://designsystem.porsche.com/latest/components/tabs-bar/examples#gradient-color
2. Tab through the tab-bar example forwards and backwards
3. after first focus of scrollable container, it is not possible to focus this container again. Therefor also the tab-items can't get focus state

### Expected behavior

Focus management should always work

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by o-hook**
### Pull Request Checklist

- [x] I have read and accept the [Contributor License Agreement](https://opensource.porsche.com/docs/cla)

#### References

- Preview: https://designsystem.porsche.com/issue/3342

#### Scope

Short description of the pull request scope

#### Resolved Issue

Resolves #3342

### CODE REVIEW

File: packages/components-js/tests/e2e/specs/tabs-bar.e2e.ts [hunk 1]
```diff
@@ -135,6 +135,32 @@ test('should work with nested or translated markup', async ({ page }) => {
...
+  expect(await getAttribute(thirdButton, 'tabindex')).toBe('-1');
+  expect(await getAttribute(thirdButton, 'aria-selected')).toBe('false');
+
+  // change active-tab-index prop
+  await page.evaluate(() => {
+    const tabsBar = document.querySelector('p-tabs-bar');
+    tabsBar.activeTabIndex = 2;
+  });
+
+  await waitForStencilLifecycle(page);
```
Comment (henrijoss) [inline packages/components-js/tests/e2e/specs/tabs-bar.e2e.ts:154]: const host = await getHost(page);
await setProperty(host, 'activeTabIndex', 2);

Something like this is how we usually do it

File: packages/components/CHANGELOG.md [hunk 2]
```diff
@@ -75,13 +75,15 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0),
...
   `TypeError: Cannot read properties of undefined (reading 'type')`
   ([#3383](https://github.com/porsche-design-system/porsche-design-system/pull/3383))
-- `LinkTile`: Broken word-break & hyphens Safari
+- `Link Tile`: Broken word-break & hyphens Safari
   ([#3397](https://github.com/porsche-design-system/porsche-design-system/pull/3397))
-- `Select-Wrapper`: `optgroup` styling and behavior
+- `Select Wrapper`: `optgroup` styling and behavior
   ([#3410](https://github.com/porsche-design-system/porsche-design-system/pull/3410))
+- `Tabs Bar`: fixed tabindex issue when `Tabs Bar` is rendered with the `Scroller` component
+  ([#3421](https://github.com/porsche-design-system/porsche-design-system/pull/3421))
```
Comment (henrijoss) [inline packages/components/CHANGELOG.md:86]: This has to be moved to unreleased

File: packages/components/src/components/tabs-bar/tabs-bar.tsx [hunk 3]
```diff
@@ -247,12 +248,13 @@ export class TabsBar {
...
-        target.tabIndex = null;
-        setTimeout(() => {
-          target.tabIndex = tabIndex;
-        });
+        if (target.matches('button')) {
+          const { tabIndex } = target;
+          target.tabIndex = null;
+          setTimeout(() => {
+            target.tabIndex = tabIndex;
+          });
```
Comment (henrijoss) [inline packages/components/src/components/tabs-bar/tabs-bar.tsx:256]: What does this do? Is it essentially just reassigning the same tabIndex value again? Maybe add a comment

Comment (o-hook) [reply packages/components/src/components/tabs-bar/tabs-bar.tsx:256]: It's not a new feature. And there is alre4ady a comment 😉.

**Review by henrijoss — APPROVED** [review]

**Pull request merged**